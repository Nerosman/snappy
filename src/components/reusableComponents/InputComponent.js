import React from "react";
import {FormGroup, Input, Label} from "reactstrap";

export const InputComponent = ({value, onChange, label, name, type, disabled, onValid, errorMessage, id, placeholder, required = false}) => (
    <FormGroup row>
        <Label for={id}>{label}</Label>
        <Input
            id={id}
            name={name}
            type={type ? type : "number"}
            placeholder={placeholder}
            value={value}
            onChange={onChange}
            disabled={disabled}
            invalid={!onValid}
            required={required}
        />
        {!onValid ? <div className="error-text">
            {errorMessage}
        </div> : null}
    </FormGroup>
);