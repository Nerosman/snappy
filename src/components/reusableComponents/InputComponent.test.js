import ReactDOM from "react-dom";
import React from "react";
import {InputComponent} from "./InputComponent";
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("InputComponent", () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<InputComponent />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders without crashing', () => {
        shallow(<InputComponent />);
    });

    it('renders welcome message', () => {
        const wrapper = shallow(<InputComponent />);
        expect(wrapper.find('.error-text').type()).toEqual('div');
    });

    test('make snapshot for InputComponent', () => {
        const wrapper = shallow(
            <InputComponent />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
