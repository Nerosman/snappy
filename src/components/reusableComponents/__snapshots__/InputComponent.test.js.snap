// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent make snapshot for InputComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        />,
        <Input
          invalid={true}
          required={false}
          type="number"
        />,
        <div
          className="error-text"
        />,
      ],
      "row": true,
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "for": undefined,
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": undefined,
          "id": undefined,
          "invalid": true,
          "name": undefined,
          "onChange": undefined,
          "placeholder": undefined,
          "required": false,
          "type": "number",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "error-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          />,
          <Input
            invalid={true}
            required={false}
            type="number"
          />,
          <div
            className="error-text"
          />,
        ],
        "row": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "for": undefined,
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": undefined,
            "id": undefined,
            "invalid": true,
            "name": undefined,
            "onChange": undefined,
            "placeholder": undefined,
            "required": false,
            "type": "number",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "error-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
